geopandas/tools/overlay.py

```diff
@@ -176,3 +177,158 @@ def overlay(df1, df2, how, use_sindex=True):
 
     # Return geodataframe with new indices
     return GeoDataFrame(collection, index=range(len(collection)))
+
+def overlay_intersection(df1, df2):
+    ''''
+    Overlay Intersection operation used in overlay function
+    '''
+    # Spatial Index to create intersections
+    spatial_index = df2.sindex
+    df1['bbox'] = df1.geometry.apply(lambda x: x.bounds)
+    df1['sidx'] = df1.bbox.apply(lambda x:list(spatial_index.intersection(x)))
+    # Create pairs of geometries in both dataframes to be intersected
+    pairs = df1['sidx'].to_dict()
+    nei = []
+    for i,j in pairs.items():
+        for k in j:
+            nei.append([i,k])
+    if nei!=[]:
+        pairs = GeoDataFrame(nei, columns=['idx1','idx2'], crs=df1.crs)
```

Comment:
you can make this a normal DataFrame, as the conversion to GeoDataFrame is done in the end